src/tests/src/api_internal/test_api.cpp

```diff
@@ -96,3 +98,35 @@ BOOST_AUTO_TEST_CASE(result_contains_problems)
     BOOST_CHECK(!results.error);
     BOOST_CHECK_EQUAL(results.antares_problems.weeklyProblems.size(), 52);
 }
+
+// Test where data in problems are consistant with data in study
+BOOST_AUTO_TEST_CASE(result_with_ortools_coin)
+{
+    Antares::API::APIInternal api;
+    auto study_loader = std::make_unique<InMemoryStudyLoader>();
+    const Antares::Solver::Optimization::OptimizationOptions opt{.ortoolsUsed = true,
+                                                                 .ortoolsSolver = "coin",
+                                                                 .solverLogs = false,
+                                                                 .solverParameters = ""};
+
+    auto results = api.run(*study_loader.get(), opt);
+
+    BOOST_CHECK(!results.antares_problems.empty());
+    BOOST_CHECK(!results.error);
+    BOOST_CHECK_EQUAL(results.antares_problems.weeklyProblems.size(), 52);
+}
+
+// Test where we use an invalid OR-Tools
+BOOST_AUTO_TEST_CASE(invalid_ortools_solver)
+{
+    Antares::API::APIInternal api;
+    auto study_loader = std::make_unique<InMemoryStudyLoader>();
+    const Antares::Solver::Optimization::OptimizationOptions opt{
+      .ortoolsUsed = true,
+      .ortoolsSolver = "this-solver-does-not-exist",
+      .solverLogs = true,
+      .solverParameters = ""};
+
+    auto shouldThrow = [&api, &study_loader, &opt] { return api.run(*study_loader.get(), opt); };
+    BOOST_CHECK_THROW(shouldThrow(), Antares::FatalError);
```

Comment:
can you wait for #2500 to be merged and use the new unit_test_utils.h to check the message of the exception ? (see examples in test_component and test_system)